week_1_2.py

```diff
@@ -0,0 +1,100 @@
+"""
+
+TASK 2
+
+Write a base class to represent a student. Below is a starter code.
+Feel free to add any more new features to your class.
+As a minimum a student has a name and age and a unique ID.
+
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student.
+
+"""
+
+
+class Student:
+
+    def __init__(self, name, age, id):
+        self.name = name
+        self.age = age
+        self.id = id
+        self.subjects = dict()
+
+    def add_subject(self, subject, grade = "N/A"):
```

Comment:
I'd advise against mixing types like this (i.e. `grade` can be _either_ an `int` or the `string` "N/A"), particularly as any non-empty string is truthy.

Instead, try to separate the ideas of data storage and presentation. You could store an absent grade as `None`, then format `None`s as `N/A` when needing to be serialized.